scripts/merge-changelogs.ts

```diff
@@ -0,0 +1,191 @@
+        version,
+        commit: match.groups?.commit || match.groups?.depsCommit,
+        type: (match.groups?.type as any) || 'Updated Dependencies',
+        message: match.groups?.message?.trim() || match.groups?.deps?.trim()
+      });
+    }
+  }
+  return results;
+}
+
+function parseChangelog(changelog: string): Message[] {
+  const packageName = getPackageName(changelog);
+  const versions = splitByVersion(changelog);
+  return versions
+    .map(({ version, content }) => parseContent(content, version, packageName))
+    .flat();
+}
+
+function writeMessagesOfType(
+  messages: Message[],
+  type: typeof validMessageTypes[number]
+): string {
+  return messages.some(msg => msg.type === type)
+    ? `## ${type}
+
+${messages
+  .filter(msg => msg.type === type)
+  .map(
+    msg => `- [${msg.packageNames.join(', ')}] ${msg.message} (${msg.commit})`
+  )
+  .join('\n')}`
+    : '';
+}
+
+function mergeMessages(parsedMessages: Message[]): Message[] {
+  return parsedMessages.reduce((prev, curr) => {
+    const sameMessage = prev.find(
+      msg =>
+        msg.message === curr.message &&
+        msg.version === curr.version &&
+        msg.type === curr.type
+    );
+    if (sameMessage) {
+      sameMessage.packageNames.push(curr.packageNames[0]);
+      return prev;
+    }
+    return [...prev, curr];
+  }, [] as Message[]);
+}
+
+function createNewSection(version: string, messages: Message[]): string {
+  return `
+# ${version}
+
+Release Date: TBD<br>
+API Docs: https://sap.github.io/cloud-sdk/api/${version}<br>
+Blog: TBD<br>
+
+${writeMessagesOfType(messages, 'Compatibility Note')}
+
+${writeMessagesOfType(messages, 'New Functionality')}
+
+${writeMessagesOfType(messages, 'Improvement')}
+
+${writeMessagesOfType(messages, 'Fixed Issue')}
+
+${
+  messages.filter(msg => msg.type === 'Updated Dependencies')
+    ? '## Updated Dependencies\n'
+    : ''
+}
+${messages
+  .filter(msg => msg.type === 'Updated Dependencies')
+  .map(
+    msg =>
+      `- [${msg.packageNames.join(', ')}] Updated Dependencies (${
+        msg.commit
+      })\n  ${msg.message}`
+  )
+  .join('\n')}
+`;
+}
+
+async function formatChangelog(parsedChangelog: Message[]): Promise<string> {
+  let unifiedChangelog = await readFile('CHANGELOG.md', { encoding: 'utf8' });
+  const relevantMessages = parsedChangelog.filter(
+    message => !unifiedChangelog.includes(`# ${message.version}`)
+  );
+  const versions = [...new Set(relevantMessages.map(msg => msg.version))];
+
+  for (const version of versions) {
+    const newContent = createNewSection(
+      version,
+      relevantMessages.filter(msg => msg.version === version)
+    );
+    unifiedChangelog =
+      unifiedChangelog.split('\n').slice(0, 30).join('\n') +
+      newContent +
+      unifiedChangelog.split('\n').slice(30).join('\n');
+  }
+
+  return unifiedChangelog;
+}
+
+async function getChangelog(): Promise<void> {
+  const files = new GlobSync('packages/*/CHANGELOG.md').found;
```

Comment:
[pp] Is ther a simple way to get the public workspaces via turborepo and then make a Promise.allSettled() to get the object just to be sure that we do not get changelogs from private packages. But I guess the changeset CLI has a filter for that so really minor thing.